Description: Don't enable use_icf with gold on mipsel
Author: Adrian Bunk <bunk@debian.org>
Forwarded: no
Last-Update: 2023-07-13

--- a/src/3rdparty/chromium/build/config/compiler/BUILD.gn
+++ b/src/3rdparty/chromium/build/config/compiler/BUILD.gn
@@ -175,11 +175,14 @@ declare_args() {
   # The bug only affects x86 and x64, so we can still use ICF when targeting
   # other architectures.
   #
+  # mipsel has address space issues (only 2 GB)
+  #
   # lld doesn't have the bug.
   use_icf = (is_posix || is_fuchsia) && !is_debug && !using_sanitizer &&
             !use_clang_coverage && !(is_android && use_order_profiling) &&
             (use_lld || (use_gold && (is_chromeos || !(current_cpu == "x86" ||
-                                                       current_cpu == "x64"))))
+                                                       current_cpu == "x64")))) &&
+            !(current_cpu == "mipsel")
 }
 
 if (is_win || is_android || (is_chromeos && is_chromeos_device)) {
