Description: apply -Os when optimize_for_size is set
Author: Adrian Bunk <bunk@debian.org>
Forwarded: no
Last-Update: 2023-07-09

--- a/src/3rdparty/chromium/build/config/compiler/BUILD.gn
+++ b/src/3rdparty/chromium/build/config/compiler/BUILD.gn
@@ -2065,6 +2065,8 @@ config("optimize") {
     #   for size by default (so, also Windows)
     # - Investigate -Oz here, maybe just for ARM?
     cflags = [ "-Os" ] + common_optimize_on_cflags
+  } else if (optimize_for_size) {
+    cflags = [ "-Os" ] + common_optimize_on_cflags
   } else {
     cflags = [ "-O2" ] + common_optimize_on_cflags
   }
@@ -2139,6 +2141,8 @@ config("optimize_max") {
       cflags = [ "/O2" ] + common_optimize_on_cflags
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
+    } else if (optimize_for_size) {
+      cflags = [ "-Os" ] + common_optimize_on_cflags
     } else {
       cflags = [ "-O2" ] + common_optimize_on_cflags
     }
@@ -2171,6 +2175,8 @@ config("optimize_speed") {
       cflags = [ "/O2" ] + common_optimize_on_cflags
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
+    } else if (optimize_for_size) {
+      cflags = [ "-Os" ] + common_optimize_on_cflags
     } else {
       cflags = [ "-O3" ] + common_optimize_on_cflags
     }
